packages/gestalt/src/SideNavigation/Collapser.js

```diff
@@ -0,0 +1,38 @@
+// @flow strict
+import { type Node as ReactNode } from 'react';
+import Box from '../Box';
+import { useSideNavigation } from '../contexts/SideNavigationProvider';
+import IconButton from '../IconButton';
+import Sticky from '../Sticky';
+
+export default function Collapser({ raised }: { raised: boolean }): ReactNode {
+  const { collapsed, overlayPreview, onCollapse, setOverlayPreview } = useSideNavigation();
+
+  return (
+    <Sticky top={0}>
+      <Box
+        display="flex"
+        justifyContent={collapsed && !overlayPreview ? 'center' : 'end'}
+        marginBottom={-2}
+        padding={2}
+        color={raised ? 'default' : undefined}
+        borderStyle={raised ? 'raisedTopShadow' : undefined}
+      >
+        <IconButton
+          icon={collapsed ? 'arrow-double-forward' : 'arrow-double-back'}
+          accessibilityLabel={
+            collapsed
+              ? 'Navigation collapsed. Click button to expand.'
+              : 'Navigation expanded. Click button to collapse.'
+          }
+          size="xs"
+          onClick={() => {
+            onCollapse?.(!collapsed);
```

Comment:
This is where the collapse state is toggled.

Reply:
This is the Collapser 
![Screenshot by Dropbox Capture](https://github.com/pinterest/gestalt/assets/10593890/568f156a-7d8d-4193-a3d9-f8dc2430cd6e)